java/src/jmri/jmrit/logix/NXFrame.java

```diff
@@ -631,16 +640,16 @@ private float getDistance(JTextField field, BlockOrder bo) throws JmriException
         if (_units.equals(Display.IN)){
             distance *= 25.4f;  // convert inches to millimeters
             if (distance < 0 || distance > pathLen) {
-                field.setText(formatter.format(pathLen * 12.07));
+                field.setText(formatter.format(pathLen * 25.4f));
                 throw new JmriException(Bundle.getMessage(
                         "BadLengthIn", bo.getPathName(), bo.getBlock().getDisplayName(), pathLen*0.039701f, text));                                        
             }
         } else {
             distance *= 10f;  // convert centimeters to millimeters
             if (distance < 0 || distance > pathLen) {
-                field.setText(formatter.format(pathLen * 5));
+                field.setText(formatter.format(pathLen * 10));
                 throw new JmriException(Bundle.getMessage(
-                        "BadLengthCm", bo.getPathName(), bo.getBlock().getDisplayName(), pathLen*0.5f, text));                                        
+                        "BadLengthCm", bo.getPathName(), bo.getBlock().getDisplayName(), pathLen*0.1f, text));                                        
```

Comment:
On line 650 you multiply with 10 instead of 5.
Here you multiply with 0.1 instead of 0.5.

Just checking that 0.1 is correct and not supposed to be 1.0.

Reply:
Fixed distance correction factor. Thanks.